tensorflow_graphics/rendering/camera/perspective.py

```diff
@@ -342,32 +365,50 @@ def project(point_3d, focal, principal_point, name=None):
         tensor=principal_point,
         tensor_name="principal_point",
         has_dim_equals=(-1, 2))
-    shape.compare_batch_dimensions(
```

Comment:
Out of curiosity, why is this `project` method (as well as `ray`) allowed to be broadcast_compatible but not the other functions?

I'm asking because, I tried to reuse the `matrix_from_intrinsics` function in the implementation of `project`, but the fact that one allows for broadcast compatibility but the other doesn't, made the testing really tricky.

Reply:
I don't think there is a really good reason. My feeling after reviewing #406 is that all these functions should coerce the skew coefficient argument to match the `focal` and `principal_point` arguments following the pattern of e.g. 
https://github.com/tensorflow/graphics/blob/3dd63357f54f42ad60c6dc0daeb3e1739f7863be/tensorflow_graphics/geometry/transformation/rotation_matrix_3d.py#L398

That should allow the default skew to be simply `0.0` and simplify the code be removing the need to handle a `None` default.

Reply:
Thanks for the pointer! I will check out the link you shared and implement the common batch shape coercion.